### PR TITLE
fix(aria-expanded) on submenues add aria prop

### PR DESCRIPTION
**Pull request by ghost**
@isidorn I am trying to figure out how to solve this aria-expended issue. 
Can you advise me that we would try to add/remove the following?
on 
<a class="action-menu-item monaco-submenu-item" **aria-expended="true">**

````
<li class="action-item focused" role="presentation">
	<a class="action-menu-item monaco-submenu-item" role="menuitem" tabindex="0" aria-checked="false" aria-haspopup="true" aria-posinset="6" aria-setsize="18" style="color: rgb(255, 255, 255); background-color: rgb(9, 71, 113);" *aria-expended="true"*>
		<span class="menu-item-check" role="none" style="background-color: rgb(255, 255, 255);"></span>
		<span class="action-label" aria-label="Open Recent">Open Recent</span>
		<span class="submenu-indicator" aria-hidden="true" style="background-color: rgb(255, 255, 255);"></span>
	</a>
</li>
```

### CODE REVIEW

**Comment by isidorn** [comment]
@georgebatalinski thanks for the PR. Currently we are in Endgame week which means we are doing testing and do not take new features - thus assigning this to September.

On the best location where we would tackle the expanded State, @sbatten would be the best since he owns the menus.
@sbatten let me know if you do not have time and I can also review / tackle this next milestone

**Comment by georgebatalinski** [comment]
@sbatten would you be able to give me some context around how the menu is structured?

this.container 
this.element 
this.item 


I am working on trying to figure out if there is no conflict with these  on the **a element**
aria-checked="false" aria-haspopup="true" *aria-expended="true"*

**Comment by georgebatalinski** [comment]
**Tests completed:**
1. Check that aria-expanded is used on elements a or button (** not sure if this is required)
2. Check that the toggling action correctly changes the state of the aria-expanded attribute
(https://www.w3.org/WAI/GL/wiki/Using_aria-expanded_to_indicate_the_state_of_a_collapsible_element)


@isidorn is it correct for an element to have both *aria-checked* and *aria-expended*?

```
<a class="action-menu-item monaco-submenu-item" role="menuitem" aria-checked="false" aria-expended="true" ...>
```

**Comment by isidorn** [comment]
@georgebatalinski according to my knowledge, `aria-checked` is used for elements which have a checkmark (toggle state). In theory an element can have both the `aria-checked` and `aria-expanded`, however I am not sure what element in vscode falls in this category? I am not aware of any menu entry that is both a checkmark and is a submenu.

Example for `aria-checked` element should be File > Auto Save
Example `aria-expanded` element should be File > Open Recent

Screenshot 2019-08-30 at 11 14 25

**Comment by georgebatalinski** [comment]
@isidorn I will review this tomorrow, thank you for the input on `aria-checked` menu elements 🚀

**Comment by georgebatalinski** [comment]
I am still working on figuring out which elements have the `aria-checked` on them 
I will spend some time debugging the `action` aspect of the menu 
in order to confirm if only the relevant menu items, get the `aria-checked`


will update soon (after fixing the build errors on my machine from yarn :)

**Comment by isidorn** [comment]
@georgebatalinski great thanks
If you can not fix those yarn errors let us know (I faced something similar on my win machine last week)

**Comment by georgebatalinski** [comment]
@isidorn this PR is ready to go - I had sometime today (so started investigating my yarn issue) 
I will created another PR for the `aria-checked`, but it should not interfere anyway, did not want to hold you up

**Comment by georgebatalinski** [comment]
gyp ERR! stack Error: spawn C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\15.0\Bin\MSBuild.exe ENOENT

**Comment by isidorn** [comment]
@georgebatalinski I think you forgot to push your latest changes to this PR, since the last change I see is from 19 days ago.

**Comment by georgebatalinski** [comment]
@isidorn the PR has been ready since then (19 days ago), 
I was just investigating something else (outside of this PR) 

I am still having yarn issues - trying to get it to work via 
https://gist.github.com/noseratio/d81888eda77620e526c7eb2c3b75cba8

**Comment by isidorn** [comment]
Looks good to me. Though leaving final review to @sbatten since he owns the menus.

**Review by isidorn — APPROVED** [review]

**Review by sbatten — APPROVED** [review]
noticed there were still 2 places we weren't setting back to false so I updated those

File: src/vs/base/browser/ui/menu/menu.ts [hunk 1]
```diff
@@ -796,13 +790,7 @@ class SubmenuMenuActionViewItem extends BaseMenuActionViewItem {
 			this.submenuDisposables.add(this.parentData.submenu.onDidCancel(() => {
 				this.parentData.parent.focus();
 
-				if (this.parentData.submenu) {
```
Comment (sbatten) [inline src/vs/base/browser/ui/menu/menu.ts]: @jeanp413 credit to you for calling this out in your PR. In this PR it became necessary to fix without further duplicating code.

**Comment by sbatten** [comment]
@georgebatalinski pinging for CLA signing

**Comment by sbatten** [comment]
@georgebatalinski are you still available to sign the CLA?

**Comment by georgebatalinski** [comment]
@sbatten yes - I will sign it

**Comment by georgebatalinski** [comment]
@sbatten and @isidorn thank you very much - I am printing this PR :) so much amazing context around how the menu system works 👍

**Pull request merged**